src/v/cluster/partition.cc

```diff
@@ -390,6 +390,23 @@ partition::timequery(storage::timequery_config cfg) {
     co_return result;
 }
 
+uint64_t partition::additional_disk_size_bytes() const {
```

Comment:
Some tests for this function would be nice, not sure if unit or ducktape tests are more convenient